### DISCUSSION TITLE
Using actions from actions

### DISCUSSION THREAD

**Discussion by amarao** [community/community#23394] [New to GitHub]
<p>I’m trying to organize a large list of similar workflows. I decided to move the common code for all workflows into local actions.</p>
<p>My code looks like this:</p>
<p>workflow:</p>
<pre><code class="lang-auto">      - name: Build
        uses: ./.github/actions/build
        with:
          param1: something
          param2: else
</code></pre>
<p>action (in <code>./gitbhu/actions/build/action.yaml</code>):</p>
<pre><code class="lang-auto">....
runs:
  using: "composite"
  steps:
   - name: build app
     run: build me ${{ input.param1}} ${{input.param2}}
   ...
   - name: Set up Docker Buildx
     id: buildx
     uses: docker/setup-buildx-action@v1.3.0
   ...
</code></pre>
<p>Turned out, I can’t use <code>uses</code> inside actions! Basically, there is no ability to reuse existing code, isn’t it?</p>
<p>How to create a large workflow then? By using aggressive copypaste? By using third party tools?</p>

**Comment by shrink** [comment, marked as answer]
<p>That’s right, at present composite actions do not support the use of other actions. There’s an open discussion about composite actions <a href="https://github.com/actions/runner/issues/646#issuecomment-860776457" rel="noopener nofollow ugc">which received a promising update last month</a>:</p>
<blockquote>
<p>I can’t give any information about timelines, but we are currently working on composite actions! :rocket: We plan on enabling the uses keyword, which allows us to support: […]</p>
</blockquote>
<p>You might also find <a href="https://github.com/github/roadmap/issues/98" rel="noopener nofollow ugc">Centrally managed workflow templates</a> on the GitHub Roadmap helpful, which is currently targeting Q3 2021.</p>
<p>If you have an absolutely critical need for reducing workflow reuse in your organisation then there are options, i.e: you could hack your way to achieving this, but based on my own experience I would <strong>highly recommend</strong> holding out until the official support is released through either of the above.</p>
